web_widget_many2one_polymorphic/README.rst

```diff
@@ -0,0 +1,74 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: https://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===========================
+Polymorphic many2one widget
+===========================
+
+The polymorphic field allows to dynamically store an id linked to any model in
+Odoo instead of the usual fixed one in the view definition, allowing comfortable
+editing in cases where you'd have to craft a reference field.
+
+A notorious case is mail.message with fields ``res_id`` and ``res_model``::
+
+    'res_model': fields.char(),
+    'res_id': fields.integer('Resource')
+
+Then in your view, you can say::
+
+    <field name="res_model" invisible="1" />
+    <field name="res_id" widget="many2one_polymorphic" options='{"model_field": "res_model"}' />
+
+This way, the user gets the full featured many2one widget for the model in question.
+
+Usage
+=====
+
+To use this module, you need to:
+
+#. depend on it
+#. use the widget in the integer field and pass the field to be used for the model in the options dictionary
```

Comment:
does it work if you populate `res_model` via onchange? You could put a select on `res_model` for instance.